# ydata-profiling checkpoint 1

**In this checkpoint, we are going to work on 'Education Attainment And Enrolment Around The World' dataset, provided by the World bank group.**

**Dataset description : The dataset was derived from the original copy, which was published on the world bank website. The primary purpose of this database is to document and analyse patterns of educational attainment rates in different country around the world.**

➡️ Dataset link

https://i.imgur.com/bMQy55e.jpg

*Columns description :*

- Age[15-49]All_gradeX : Percentage of the entire population ages 15 to 49 that has completed grade X
- Age[15-49]Male_gradeX : Percentage of the male population ages 15 to 49 that has completed grade X
- Age[15-49]Female_gradeX : Percentage of the female population ages 15 to 49 that has completed grade X
- X = [1...3]

**Instructions**

**Step 1**: Data exploration with Pandas
1. Download the provided dataset
2. Use pandas to read the file into a pandas Dataframe.
3. Display general information about the dataset.
4. Run descriptive analysis on the dataset using the 'describe' function.

**Step 2**: Data exploration with ydata-profiling
1. Use ydata-profiling to generate a report of the provided dataset.
2. Look for missing values in the dataset. You can use the report generated by ydata to identify missing values.
3. Examine the correlations between different columns in the dataset. You can use the report generated by ydata to identify correlations.
4. Identify any other interesting patterns or insights in the dataset that you can discover through the ydata-profilingreport.

**Step 3**: Summary
1. Write a summary of your findings based on the ydata-profiling report. Be sure to include any interesting patterns or insights you discovered in the dataset.

### Data exploration with Pandas

In [1]:
import pandas as pd

In [2]:
#Using pandas to read the file into a pandas Dataframe
df = pd.read_csv('EDUCATION_ATTAINMENT.csv')
pd.set_option('display.max_columns', None)

In [3]:
df.head()

Ages[15-49]_All_grade1  Ages[15-49]_All_grade2  Ages[15-49]_All_grade3  \
0                0.669091                0.667214                0.651812   
1                0.485879                0.480171                0.465510   
2                0.516431                0.514233                0.502269   
3                0.914294                0.903528                0.871690   
4                0.876990                0.841497                0.762864   

   Ages[20-29]_All_grade1  Ages[20-29]_All_grade2  Ages[20-29]_All_grade3  \
0                0.460899                0.458749                0.446925   
1                0.267236                0.265735                0.257999   
2                0.327775                0.326250                0.318003   
3                0.853770                0.840792                0.810794   
4                0.881265                0.854158                0.795892   

   Ages[30-39]_All_grade1  Ages[30-39]_All_grade2  Ages[30-39]_All_grade3  \
0                0.292791                0.290929                0.279100   
1                0.191279                0.188839                0.182693   
2                0.221981                0.220472                0.214040   
3                0.775725                0.757109                0.713569   
4                0.883625                0.853812                0.805231   

   Ages[40-49]_All_grade1  Ages[40-49]_All_grade2  Ages[40-49]_All_grade3  \
0                0.250981                0.248945                0.237003   
1                0.187151                0.186541                0.183361   
2                0.205611                0.205062                0.200147   
3                0.786468                0.758427                0.708025   
4                0.747516                0.717519                0.646261   

   Ages[15-49]_Male_grade1  Ages[15-49]_Male_grade2  Ages[15-49]_Male_grade3  \
0                 0.816546                 0.814355                 0.801192   
1                 0.615897                 0.610735                 0.594344   
2                 0.645472                 0.643722                 0.632224   
3                 0.943719                 0.934511                 0.901751   
4                 0.912438                 0.882079                 0.805788   

   Ages[20-29]_Male_grade1  Ages[20-29]_Male_grade2  Ages[20-29]_Male_grade3  \
0                 0.644394                 0.641686                 0.625674   
1                 0.417831                 0.415445                 0.404348   
2                 0.488213                 0.486835                 0.477032   
3                 0.917778                 0.911683                 0.895311   
4                 0.947246                 0.931045                 0.901097   

   Ages[30-39]_Male_grade1  Ages[30-39]_Male_grade2  Ages[30-39]_Male_grade3  \
0                 0.467792                 0.466380                 0.448260   
1                 0.305830                 0.301790                 0.292361   
2                 0.356104                 0.354028                 0.344222   
3                 0.869510                 0.857323                 0.831267   
4                 0.967384                 0.954881                 0.929648   

   Ages[40-49]_Male_grade1  Ages[40-49]_Male_grade2  Ages[40-49]_Male_grade3  \
0                 0.398761                 0.395617                 0.377484   
1                 0.309030                 0.307808                 0.302900   
2                 0.330364                 0.329344                 0.321027   
3                 0.888346                 0.871305                 0.843261   
4                 0.913294                 0.894267                 0.836943   

   Ages[15-49]_Female_grade1  Ages[15-49]_Female_grade2  \
0                   0.512733                   0.511190   
1                   0.337259                   0.330927   
2                   0.368800                   0.3

In [4]:
#Displaying general information about the dataset.

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 626 entries, 0 to 625
Data columns (total 38 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Ages[15-49]_All_grade1     626 non-null    float64
 1   Ages[15-49]_All_grade2     626 non-null    float64
 2   Ages[15-49]_All_grade3     626 non-null    float64
 3   Ages[20-29]_All_grade1     626 non-null    float64
 4   Ages[20-29]_All_grade2     626 non-null    float64
 5   Ages[20-29]_All_grade3     626 non-null    float64
 6   Ages[30-39]_All_grade1     625 non-null    float64
 7   Ages[30-39]_All_grade2     625 non-null    float64
 8   Ages[30-39]_All_grade3     625 non-null    float64
 9   Ages[40-49]_All_grade1     625 non-null    float64
 10  Ages[40-49]_All_grade2     625 non-null    float64
 11  Ages[40-49]_All_grade3     625 non-null    float64
 12  Ages[15-49]_Male_grade1    626 non-null    float64
 13  Ages[15-49]_Male_grade2    626 non-null    float64

In [5]:
#Running descriptive analysis on the dataset using the 'describe' function.

df.describe().T

count         mean       std          min  \
Ages[15-49]_All_grade1     626.0     0.873886  0.162624     0.251917   
Ages[15-49]_All_grade2     626.0     0.861490  0.166895     0.244071   
Ages[15-49]_All_grade3     626.0     0.837509  0.176555     0.233274   
Ages[20-29]_All_grade1     626.0     0.823849  0.200559     0.079610   
Ages[20-29]_All_grade2     626.0     0.812738  0.202587     0.079610   
Ages[20-29]_All_grade3     626.0     0.789965  0.207979     0.079610   
Ages[30-39]_All_grade1     625.0     0.766171  0.236563     0.098059   
Ages[30-39]_All_grade2     625.0     0.752519  0.237679     0.091358   
Ages[30-39]_All_grade3     625.0     0.724980  0.240681     0.085236   
Ages[40-49]_All_grade1     625.0     0.709166  0.259998     0.036549   
Ages[40-49]_All_grade2     625.0     0.692149  0.260356     0.033880   
Ages[40-49]_All_grade3     625.0     0.659274  0.260564     0.031706   
Ages[15-49]_Male_grade1    626.0     0.894628  0.136706     0.317084   
Ages[15-49]_Male_grade2    626.0     0.881461  0.141548     0.314678   
Ages[15-49]_Male_grade3    626.0     0.855987  0.152427     0.277139   
Ages[20-29]_Male_grade1    626.0     0.864598  0.160596     0.164837   
Ages[20-29]_Male_grade2    626.0     0.854039  0.162849     0.164837   
Ages[20-29]_Male_grade3    626.0     0.831923  0.169123     0.164837   
Ages[30-39]_Male_grade1    625.0     0.819950  0.194460     0.140272   
Ages[30-39]_Male_grade2    625.0     0.807381  0.196286     0.128866   
Ages[30-39]_Male_grade3    625.0     0.781135  0.200962     0.121611   
Ages[40-49]_Male_grade1    625.0     0.772520  0.225644     0.047596   
Ages[40-49]_Male_grade2    625.0     0.757116  0.226886     0.043235   
Ages[40-49]_Male_grade3    625.0     0.726091  0.229169     0.040175   
Ages[15-49]_Female_grade1  626.0     0.853510  0.192017     0.165652   
Ages[15-49]_Female_grade2  626.0     0.841888  0.196022     0.165652   
Ages[15-49]_Female_grade3  626.0     0.819418  0.204955     0.159859   
Ages[20-29]_Female_grade1  626.0     0.790544  0.236541     0.029662   
Ages[20-29]_Female_grade2  626.0     0.778972  0.238637     0.029662   
Ages[20-29]_Female_grade3  626.0     0.755644  0.243755     0.029662   
Ages[30-39]_Female_grade1  625.0     0.719020  0.278133     0.060930   
Ages[30-39]_Female_grade2  625.0     0.704342  0.279059     0.058368   
Ages[30-39]_Female_grade3  625.0     0.675526  0.281139     0.053241   
Ages[40-49]_Female_grade1  625.0     0.646864  0.302723     0.022547   
Ages[40-49]_Female_grade2  625.0     0.628309  0.302637     0.022022   
Ages[40-49]_Female_grade3  625.0     0.593585  0.301434     0.020972   
year                       626.0  2005.049521  7.894248  1981.000000   

                                   25%          50%          75%          max  
Ages[15-49]_All_grade1        0.831791     0.948926     0.984447     1.000000  
Ages[15-49]_All_grade2        0.803131     0.935167     0.980112     1.000000  
Ages[15-49]_All_grade3        0.757465     0.909587     0.971166     1.000000  
Ages[20-29]_All_grade1        0.744915     0.910132     0.973777     1.000000  
Ages[20-29]_All_grade2        0.723418     0.896568     0.967410     0.999824  
Ages[20-29]_All_grade3        0.678869     0.864849     0.955625     0.999264  
Ages[30-39]_All_grade1        0.639447     0.853861     0.959097     1.000000  
Ages[30-39]_All_grade2        0.612464     0.831246     0.948259     1.000000  
Ages[30-39]_All_grade3        0.561545     0.794898     0.922841     1.000000  
Ages[40-49]_All_grade1        0.527055     0.785356     0.937129     1.000000  
Ages[40-49]_All_grade2        0.502672     0.760970     0.923447     0.999517  
Ages[40-49]_All_grade3        0.461736     0.711113     0.887080     0.999484  
Ages[15-49]_Male_grade1       0.868194     0.951003     0.985329     1.000000  
Ages[15-49]_Male_grade2       0.833089     0.937148     0.980045     1.000000  
Ages[15-49]_Male_grade3       0.789928     0.910529     0.971076     1.000000  
Ages[20-29

In [6]:
randomSample = df.sample(n = 300)

### Data exploration with ydata-profiling

In [7]:
from ydata_profiling import ProfileReport

In [8]:
profile = ProfileReport(randomSample, title='Education Attainment And Enrolment Around The World', explorative=True)

In [ ]:
profile.to_notebook_iframe()

### Summary

The dataset provides a detailed analysis of grades for different age groups (15-49, 20-29, 30-39, and 40-49) across genders (male and female) and their distribution over the years. Here are the key insights:

1. **Overall Trends:**
   - The grades for all age groups and genders are relatively high, with means generally above 0.7.
   - There is a noticeable trend where the younger age groups tend to have higher grades compared to the older age groups.

2. **Age Group Comparisons:**
   - **Ages 15-49:** This group shows the highest mean grades across all three categories (grade 1, grade 2, and grade 3). Grade 1 has the highest mean (0.8739), followed by grade 2 (0.8615), and grade 3 (0.8375).
   - **Ages 20-29:** This age group also shows high grades but slightly lower than the 15-49 age group. The mean grades are around 0.79 to 0.82.
   - **Ages 30-39:** The grades start to decline in this group with means ranging from 0.72 to 0.77.
   - **Ages 40-49:** This group has the lowest grades among all age groups, with means ranging from 0.66 to 0.71.

3. **Gender Comparisons:**
   - **Males:** Across all age groups, males tend to have slightly higher grades than females. For instance, in the 15-49 age group, the mean grade for males is 0.8946 (grade 1), while for females, it is 0.8535.
   - **Females:** Although females have slightly lower grades than males, their grades are still high, with means generally above 0.6 for all age groups.

4. **Distribution and Variability:**
   - The standard deviations are relatively consistent within each age group and gender, indicating similar variability in grades.
   - The minimum grades show significant variability, especially in the older age groups (e.g., Ages 40-49 with a minimum of around 0.03).

5. **Yearly Distribution:**
   - The year distribution indicates that the dataset spans from 1981 to 2020, with a mean year of approximately 2005. This suggests a broad temporal coverage, allowing for longitudinal analysis.

Overall, the dataset reveals that younger age groups and males tend to have higher grades, and there is a general decline in grades as age increases. The consistency in variability suggests stable grading patterns across different demographics.